hazelcast/src/main/java/com/hazelcast/nio/tcp/TcpIpConnectionManager.java

```diff
@@ -304,9 +304,9 @@ private boolean checkAlreadyConnected(TcpIpConnection connection, Address remote
             if (existingConnection != connection) {
                 if (logger.isFinestEnabled()) {
                     log(Level.FINEST, existingConnection + " is already bound to " + remoteEndPoint
-                            + ", new one is " + connection);
+                            + ", destroying the new " + connection);
                 }
-                activeConnections.add(connection);
+                destroyConnection(connection);
```

Comment:
well, this shouldn't be a side-effect of this method. perhaps a caller should close the connection when this returns true.